bin/setup_venv.sh

```diff
@@ -265,20 +267,15 @@ else
   chmod +x "${VIRTUAL_ENV}/bin/kubectl"
 fi
 
-#
 # Download Pulumi CLI tooling
-#
-# Added some error checking to handle a failure here since it can be a bit confusing to the user (or at least to me)
-# when this bombs out and you can't figure out why. Note that the logic is specifically looking for "~=" in the
-# requirements file. Other comparisons that are legal in terms of the requirements file WILL fail with this logic.
-#
+# Regular expression and sed command from https://superuser.com/a/363878
 echo "Downloading Pulumi CLI into virtual environment"
-PULUMI_VERSION="$(grep '^pulumi~=.*$' "${script_dir}/../pulumi/python/requirements.txt" | cut -d '=' -f2 || true)"
+PULUMI_VERSION="$(pip list | grep 'pulumi ' | sed -nre 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p')"
```

Comment:
**annotation:**
Here we read the pulumi version from pip to get the version that was actually installed.